src/main/java/core/basesyntax/service/impl/ParserServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.FruitTransactionService;
+import core.basesyntax.service.ParserService;
+import java.util.List;
+
+public class ParserServiceImpl implements ParserService {
+    private final int operationCodeIndex;
+    private final int quantityIndex;
+    private final int fruitNameIndex;
+
+    private final FruitTransactionService fruitTransactionService;
+
+    public ParserServiceImpl() {
+        operationCodeIndex = 0;
+        quantityIndex = 2;
+        fruitNameIndex = 1;
+        fruitTransactionService = new FruitTransactionServiceImpl();
+    }
+
+    @Override
+    public void parseData(List<String> inputDataRows) {
```

Comment:
This method could return a List of FruitTransactions. This helps testing application in future and reveal understandable order in Main.